flatsurf/geometry/categories/similarity_surfaces.py

```diff
@@ -444,6 +444,387 @@ class Oriented(SurfaceCategoryWithAxiom):
+
+                    sage: from flatsurf import Polygon, similarity_surfaces
+                    sage: P = Polygon(vertices=[(0, 0), (2, 0), (1, 4), (0, 5)])
+                    sage: S = similarity_surfaces.self_glued_polygon(P)
+                    sage: S(0, 0).angle()
+                    1
+                    sage: S(0, (1, 0)).angle()
+                    1/2
+                    sage: S(0, (1, 0)).angle(numerical=True)
+                    0.500000000000000
+
+                TESTS:
+
+                For mutable surfaces, the angles are not cached::
+
+                    sage: from flatsurf import MutableOrientedSimilaritySurface
+                    sage: T = MutableOrientedSimilaritySurface.from_surface(S)
+                    sage: T._refine_category_(S.category())  # make angle() available
+                    sage: T(0, 0).angle()
+                    1
+                    sage: T.glue((0, 0), (0, 2))
+                    sage: T(0, 0).angle()
+                    3/8
+
+                .. SEEALSO::
+
+                    :meth:`turns` for the number of full 2π turns at a point.
+
+                """
+                if self.is_vertex():
+                    angle = self._angle_vertex(numerical=numerical)
+
+                    if self.parent().is_mutable():
+                        self._angle_vertex.clear_cache()
+
+                    return angle
+
+                from sage.all import QQ, RR
+                ring = RR if numerical else QQ
+
+                if self.is_in_edge_interior():
+                    label, coordinates = self.representative()
+                    edge = self.parent().polygon(label).get_point_position(coordinates).get_edge()
+
+                    opposite_edge = self.parent().opposite_edge(label, edge)
+                    if opposite_edge is None or opposite_edge == (label, edge):
+                        # The total angle at a self-glued or unglued edge is π.
+                        return ring(.5)
+
+                return ring.one()
+
+            @cached_in_parent_method
+            def _angle_vertex(self, numerical):
+                r"""
+                Return the total angle at this vertex in multiples of 2π.
+
+                INPUT:
+
+                - ``numerical`` -- a boolean; whether to return the angle as a
+                  floating point number.
+
+                EXAMPLES::
+
+                    sage: from flatsurf import Polygon, similarity_surfaces
+                    sage: P = Polygon(vertices=[(0, 0), (2, 0), (1, 4), (0, 5)])
+                    sage: S = similarity_surfaces.self_glued_polygon(P)
+                    sage: S(0, 0)._angle_vertex(numerical=False)
+                    1
+
+                """
+                turns, start, end = self.turns()
+
+                from flatsurf.geometry.euclidean import angle
+                return turns + angle(start, end, numerical=numerical)
+
+            def turns(self, start=None, start_holonomy=None):
```

Comment:
@videlec is it reasonable to call this parameter `start_holonomy`?

Reply:
What is `def turns`?

Reply:
Rename `start` to `start_edge` and clarify that it's outgoing.